### ISSUE TITLE
Suppress `Style/RegexpLiteral` when otherwise `AmbiguousRegexpLiteral`?

### ISSUE THREAD

**Issue by yheuhtozr** [rubocop/rubocop#13330]
## Is your feature request related to a problem? Please describe.

``` ruby
obj.method /^Regexp/ # <= triggers Lint/AmbiguousRegexpLiteral
# But...
obj.method %r{^Regexp} # <= triggers Style/RegexpLiteral
# Therefore only...
obj.method(/^Regexp/) # is warning-free
```

## Describe the solution you'd like

Is it possible to relax `Style/RegexpLiteral` so that it won't complain when using the normal Regexp literal would invoke `AmbiguousRegexpLiteral`?

## Describe alternatives you've considered

To add an config option for `Style/RegexpLiteral` to the same effect.

## Additional context

Currently, the linter only allows it when the normal Regexp literal itself creates a syntax error.
https://rubydoc.info/gems/rubocop/RuboCop/Cop/Style/RegexpLiteral
> NOTE: The following `%r` cases using a regexp starts with a blank or `=` as a method argument allowed to prevent syntax errors.

(notation corrected by me)

I am not aware of existence of a general design principle that prefers primary literals over parentheses omission for final arguments, but correct me if I am wrong.

**Comment by dvandersluis** [comment]
If you don't want to enforce parentheses around method arguments, you should configure `Style/MethodCallWithArgsParentheses` with `EnforcedStyle: omit_parentheses`. This will then allow `%r` regexp literals in `Style/RegexpLiteral` without offense:

```sh
$ cat test.rb
obj.method %r{^Regexp}


$ cat .rubocop.yml
Style/RegexpLiteral:
  EnforcedStyle: slashes

Style/MethodCallWithArgsParentheses:
  EnforcedStyle: omit_parentheses


$ rubocop test.rb -A
Inspecting 1 file
.

1 file inspected, no offenses detected
```

If you want parentheses around methods, RuboCop's autocorrect will add them in this case which does not create an infinite loop:

```sh
$ cat test.rb
obj.method %r{^Regexp}


$ cat .rubocop.yml
Style/RegexpLiteral:
  EnforcedStyle: slashes

Style/MethodCallWithArgsParentheses:
  EnforcedStyle: require_parentheses


$ rubocop test.rb -A
Inspecting 1 file
W

Offenses:

test.rb:3:12: W: [Corrected] Lint/AmbiguousRegexpLiteral: Ambiguous regexp literal. Parenthesize the method arguments if it's surely a regexp literal, or add a whitespace to the right of the / if it should be a division.
obj.method /^Regexp/
           ^
test.rb:3:12: C: [Corrected] Style/RegexpLiteral: Use // around regular expression.
obj.method %r{^Regexp}
           ^^^^^^^^^^^

1 file inspected, 2 offenses detected, 2 offenses corrected


$ cat test.rb  (after correction)
obj.method(/^Regexp/)
```

Therefore there does not seem to be any bug.
